src/Illuminate/Http/Resources/Json/ResourceCollection.php

```diff
@@ -67,8 +86,25 @@ public function toArray($request)
      */
     public function toResponse($request)
     {
-        return $this->resource instanceof AbstractPaginator
-                    ? (new PaginatedResourceResponse($this))->toResponse($request)
-                    : parent::toResponse($request);
+        if ($this->resource instanceof AbstractPaginator) {
+            return $this->preparePaginatedResponse($request);
+        }
+
+        return parent::toResponse($request);
+    }
+
+    /**
+     * Create an paginate-aware HTTP response.
+     *
+     * @param \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\JsonResponse
+     */
+    private function preparePaginatedResponse($request)
```

Comment:
It's okay to have this as method as `protected`. I think.

Reply:
Force of habit, I guess 😄